### PR TITLE
[Android] Search filters

### PR DESCRIPTION
**Pull request by surajsau**
## Issue
- #600 
- #595 

## Overview
- Added MaterialDesign package to `:feature:sessions` for `BottomSheetScaffold`
- Added `selectAllCategories` query in `TimetableItemSession.sq` to fetch all unique categories to be shown in Categories filter.
- Made the filter section horizontally scrollable since while selecting multiple categories will lead to a longer category filter button.

## Screenshot

https://user-images.githubusercontent.com/5493586/191459642-5ef99fd7-e16b-4e45-9d93-b2e8c55ae180.mp4

### CODE REVIEW

File: feature/sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/SearchUiModel.kt [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
+        get() = selectedDay != null
+
+    val selectedCategoriesValue: String
+        get() = selectedCategories.joinToString { it.title.currentLangTitle }
+
+    val isCategoriesSelected: Boolean
+        get() = selectedCategories.isNotEmpty()
+}
+
+sealed interface SearchFilterSheetState {
```
Comment (takahirom) [inline feature/sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/SearchUiModel.kt:29]: 🆒

File: feature/sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/FilterButton.kt [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+@Composable
+fun FilterButton(
+    isSelected: Boolean,
+    isDropDown: Boolean,
+    text: String,
+    modifier: Modifier = Modifier,
+    onClicked: () -> Unit
+) {
+    val transition = updateTransition(targetState = isSelected)
```
Comment (takahirom) [inline feature/sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/FilterButton.kt:32]: 🆒

**Review by takahirom — APPROVED** [review]
Looks great!

**Comment by surajsau** [comment]
Thanks! Merging this since it is approved ❤️

**Pull request merged**